bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/AuthHandler.java

```diff
@@ -307,20 +297,18 @@ public void messageReceived(ChannelHandlerContext ctx,
         }
 
         @Override
-        public void writeRequested(ChannelHandlerContext ctx,
-                                   MessageEvent e)
-                throws Exception {
+        public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
             synchronized (this) {
                 if (authenticated) {
-                    super.writeRequested(ctx, e);
-                } else if (e.getMessage() instanceof BookkeeperProtocol.Request) {
+                    super.write(ctx, msg, promise);
+                } else if (msg instanceof BookkeeperProtocol.Request) {
                     // let auth messages through, queue the rest
-                    BookkeeperProtocol.Request req = (BookkeeperProtocol.Request)e.getMessage();
+                    BookkeeperProtocol.Request req = (BookkeeperProtocol.Request) msg;
                     if (req.getHeader().getOperation()
                             == BookkeeperProtocol.OperationType.AUTH) {
-                        super.writeRequested(ctx, e);
+                        super.write(ctx, msg, promise);
                     } else {
-                        waitingForAuth.add(e);
+                        waitingForAuth.add(msg);
                     }
                 } // else just drop
```

Comment:
I think the implementation of the auth provider here was to accumulate all the responses while the auth handshake is being performed.